harness/determined/_trial_context.py

```diff
@@ -14,13 +14,17 @@ class TrialContext:
     def __init__(
         self,
         core_context: core.Context,
-        env: det.EnvContext,
+        env: Optional[det.EnvContext] = None,
```

Comment:
Maybe PyTorchTrialContext should just skip calling `super().__init__()` now.  Eventually there will only be PyTorchTrialContext and the others will be `rm -rf`'d.

But for now, this one `__init__()` that is trying to serve the needs of three trials that are increasingly different is looking more and more like a mistake.

Reply:
done.